packages/adapters/wagmi/src/client.ts

```diff
@@ -1012,10 +1004,6 @@ export class WagmiAdapter implements ChainAdapter {
         }
       })
 
-      provider.onIsConnected(() => {
-        provider.connect()
-      })
```

Comment:
Whenever we check if we're connected we call `provider.connect`, which resulted into more excessive network calls

Reply:
We should check if this is required for previously conected connections I think

Reply:
I wonder why we have this only for wagmi client